### PR TITLE
Feature/form store

### PR DESCRIPTION
**Pull request by Landon696**
## Status
Ready

## Description
Add Form store, check if network is avaible before posting a feature

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Submit forms

```sh
git fetch --all
git checkout <feature_branch> 
./gradlew connectedCheck
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/dataAdapter/GeoPackageAdapter.java [hunk 1]
```diff
@@ -482,22 +482,25 @@ public void call(Subscriber<? super SCSpatialFeature> subscriber) {
 
     private void postCreatedFeature(SCSpatialFeature feature) {
         String formId = "";
-        for(SCFormConfig config : SpatialConnect.getInstance().getDataService().getDefaultStore().getFormConfigs()) {
+        for(SCFormConfig config : SpatialConnect.getInstance().getDataService().getFormStore().getFormConfigs()) {
```
Comment (georific) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/dataAdapter/GeoPackageAdapter.java]: Is there a way to get the fromId from the SCSpatialFeature object

Comment (georific) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/dataAdapter/GeoPackageAdapter.java]: move up to FormStore layer

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/FormStore.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import rx.Observable;
+import rx.functions.Action0;
+
+/**
```
Comment (marcusthebrown) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/FormStore.java]: @georific can you add the license to the top of the file before the package statement?

Comment (Landon696) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/FormStore.java]: done

Comment (georific) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/FormStore.java]: done

**Comment by marcusthebrown** [comment]
👍

**Pull request merged**